src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1676,11 +1692,26 @@ public override BoundNode VisitNullCoalescingOperator(BoundNullCoalescingOperato
                     throw ExceptionUtilities.UnexpectedValue(node.OperatorResultKind);
             }
 
-            bool? resultIsNullable = getIsNullable(leftOperand, leftResult) == false ? false : getIsNullable(rightOperand, rightResult);
-            _resultType = TypeSymbolWithAnnotations.Create(resultType, resultIsNullable);
+            NullableAnnotation resultNullableAnnotation = getNullableAnnotation(leftOperand, leftResult);
+            
+            if (!resultNullableAnnotation.IsAnyNotNullable())
```

Comment:
>IsAnyNotNullable [](start = 42, length = 16)

Feels like this should be `!resultNullableAnnotation.ValueCanBeNull()` instead. `IsAnyNotNullable()` checks the NotAnnotated case, but without looking at `IsPossiblyNullableReferenceTypeTypeParameter`. 

A scenario that may be affected: `boolValue ? tValue : new T()`. Unless we have some kind of invariant about annotations in the result type in NullableWalker, the result type for `tValue` may be un-annotated (NotAnnotated) but actually nullable once we also look at the type.

Reply:
> Feels like this should be !resultNullableAnnotation.ValueCanBeNull() instead. IsAnyNotNullable() checks the NotAnnotated case, but without looking at IsPossiblyNullableReferenceTypeTypeParameter. 

There is no API called ```ValueCanBeNull``` on ```NullableAnnotation```. Also, the ```resultNullableAnnotation``` already contains effective annotation

---
In reply to: [231646064](https://github.com/dotnet/roslyn/pull/30913#discussion_r231646064) [](ancestors = 231646064)

Reply:
Ah, thanks. We're already using `GetValueNullableAnnotation` inside `getNullableAnnotation` local. That works

---
In reply to: [231658467](https://github.com/dotnet/roslyn/pull/30913#discussion_r231658467) [](ancestors = 231658467,231646064)